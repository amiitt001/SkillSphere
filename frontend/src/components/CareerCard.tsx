import React from 'react';

type CareerCardProps = {
  title: string;
  justification: string;
  roadmap: string[];
};

const CareerCard = ({ title, justification, roadmap }: CareerCardProps) => {
  return (
    <div className="bg-slate-800 p-6 rounded-lg shadow-lg">
      <h3 className="text-xl font-bold text-sky-400 mb-2">{title}</h3>
      <p className="text-slate-400 mb-4">{justification}</p>
      <div>
        <h4 className="font-semibold text-white mb-2">Roadmap:</h4>
        <ul className="list-disc list-inside space-y-2 text-slate-400">
          {roadmap.map((step, index) => (
            <li key={index}>{step}</li>
          ))}
        </ul>
      </div>
    </div>
  );
};

export default CareerCard;