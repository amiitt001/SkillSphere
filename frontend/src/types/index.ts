export interface Recommendation {
  title: string;
  justification: string;
  roadmap: string[];
}
