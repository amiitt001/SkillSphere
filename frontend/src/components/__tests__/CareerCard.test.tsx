import React from 'react';
import { render } from '@testing-library/react';
import CareerCard from '../CareerCard';
import { Recommendation } from '@/types';

// This is the main test block for the CareerCard component
describe('CareerCard Component', () => {

  // Create some sample data to pass to the component
  const mockRecommendation: Recommendation = {
    title: "Test Career",
    justification: "This is a test justification.",
    roadmap: ["Step 1", "Step 2"],
    estimatedSalary: "â‚¹10,00,000 LPA",
    suggestedCertifications: ["Cert A", "Cert B"],
    keyCompanies: ["Company X", "Company Y"]
  };

  // This is our first test case
  it('renders correctly and matches snapshot', () => {
    // 1. "render" the component with our sample data
    const { asFragment } = render(<CareerCard {...mockRecommendation} />);
    
    // 2. Take a "snapshot" of the rendered HTML
    // The first time you run this, it will create a snapshot file.
    // Every time after, it will compare the component's output to that saved snapshot.
    expect(asFragment()).toMatchSnapshot();
  });

});
