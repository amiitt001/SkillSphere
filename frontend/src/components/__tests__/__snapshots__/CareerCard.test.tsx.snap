// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`CareerCard Component renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-slate-800 p-6 rounded-lg shadow-lg flex flex-col"
  >
    <h3
      class="text-xl font-bold text-sky-400 mb-2"
    >
      Test Career
    </h3>
    <p
      class="text-slate-400 mb-4 text-sm"
    >
      This is a test justification.
    </p>
    <div
      class="border-t border-slate-700 pt-4 mt-auto"
    >
      <h4
        class="font-semibold text-white mb-2 text-sm"
      >
        Roadmap:
      </h4>
      <ul
        class="list-disc list-inside space-y-2 text-slate-400 text-sm mb-4"
      >
        <li>
          Step 1
        </li>
        <li>
          Step 2
        </li>
      </ul>
      <div
        class="flex items-start gap-3 mt-4"
      >
        <svg
          class="w-5 h-5"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 6v12m-3-2.818.879.659c1.171.879 3.07.879 4.242 0 1.172-.879 1.172-2.303 0-3.182C13.536 12.219 12.768 12 12 12c-.725 0-1.45-.22-2.003-.659-1.106-.879-1.106-2.303 0-3.182s2.9-.879 4.006 0l.415.33M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <div>
          <h5
            class="font-semibold text-white text-sm"
          >
            Salary
          </h5>
          <p
            class="text-slate-400 text-sm"
          >
            â‚¹10,00,000 LPA
          </p>
        </div>
      </div>
      <div
        class="flex items-start gap-3 mt-4"
      >
        <svg
          class="w-5 h-5"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4.26 10.147a60.436 60.436 0 00-.491 6.347A48.627 48.627 0 0112 20.904a48.627 48.627 0 018.232-4.41 60.46 60.46 0 00-.491-6.347m-15.482 0a50.57 50.57 0 00-2.658-.813A59.905 59.905 0 0112 3.493a59.905 59.905 0 0110.399 5.84c-.896.248-1.783.52-2.658.814m-15.482 0l-.07.002z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <div>
          <h5
            class="font-semibold text-white text-sm"
          >
            Certifications
          </h5>
          <p
            class="text-slate-400 text-sm"
          >
            Cert A, Cert B
          </p>
        </div>
      </div>
      <div
        class="flex items-start gap-3 mt-4"
      >
        <svg
          class="w-5 h-5"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2.25 21h19.5m-18-18v18m10.5-18v18m6-13.5V21M6.75 6.75h.75m-.75 3h.75m-.75 3h.75m3-6h.75m-.75 3h.75m-.75 3h.75M6.75 21v-3.375c0-.621.504-1.125 1.125-1.125h2.25c.621 0 1.125.504 1.125 1.125V21M3 3h12m6.75 3.375V3"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <div>
          <h5
            class="font-semibold text-white text-sm"
          >
            Key Companies
          </h5>
          <p
            class="text-slate-400 text-sm"
          >
            Company X, Company Y
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
